source/main/physics/Beam.h

```diff
@@ -474,6 +474,7 @@ class Actor : public ZeroedMemoryAllocator
     bool              m_antilockbrake;         //!< GUI state
     bool              m_tractioncontrol;       //!< GUI state
     bool              m_ongoing_reset;         //!< Hack to prevent position/rotation creep during interactive truck reset
+    bool              m_first_physics_step;    //!< Hack to prevent submesh destruction

```

Comment:
Good comment, but what exactly is at risk of destruction?

Reply:
The inter-vehicle-collision code needs to run after `CalcBeams` and before `CalcNodes`. In order to achieve that without a massive performance penalty I had to reorder the execution of them, which breaks the intra-vehicle-collision code (for some vehicles) without this hack.